### PR TITLE
Inference C generator

### PR DESCRIPTION
**Pull request by kiritigowda**
valid input tensor check.

### CODE REVIEW

File: vx_nn/src/batch_normalization_layer.cpp [hunk 1]
```diff
@@ -138,7 +138,19 @@ static vx_status VX_CALLBACK initializeBatchNormalizationLayer(vx_node node, con
...
+        ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[4], VX_TENSOR_BUFFER_OPENCL, &data->bnBias, sizeof(data->bnBias)));
+    }
+    else{
+        vx_context   vxContext = vxGetContext((vx_reference)node);
+        cl_context context;
+        ERROR_CHECK_STATUS(vxQueryContext(vxContext, VX_CONTEXT_ATTRIBUTE_AMD_OPENCL_CONTEXT, &context, sizeof(context)));
+        cl_float pattern = 0;
+        data->bnBias = clCreateBuffer(context, CL_MEM_READ_WRITE, sizeof(float)*input_dims[2], NULL, NULL);
+        cl_int err = clEnqueueFillBuffer(data->handle->cmdq, data->bnBias, &pattern, sizeof(cl_float), 0, input_dims[2], 0, NULL, NULL);
+        if (err) return VX_FAILURE;
```
Comment (rgiduthuri) [inline vx_nn/src/batch_normalization_layer.cpp]: When parameter[4] is not specified, there will be a memory leak as there is no matching release corresponding to clCreateBuffer.

File: vx_nn/src/scale_layer.cpp [hunk 2]
```diff
@@ -119,7 +119,18 @@ static vx_status VX_CALLBACK initializeScaleLayer(vx_node node, const vx_referen
...
+    if(parameters[2]){
+        ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_BUFFER_OPENCL, &data->bnBias, sizeof(data->bnBias)));
+    }
+    else{
+        vx_context   vxContext = vxGetContext((vx_reference)node);
+        cl_context context;
+        ERROR_CHECK_STATUS(vxQueryContext(vxContext, VX_CONTEXT_ATTRIBUTE_AMD_OPENCL_CONTEXT, &context, sizeof(context)));
+        cl_float pattern = 0;
+        data->bnBias = clCreateBuffer(context, CL_MEM_READ_WRITE, sizeof(float)*input_dims[2], NULL, NULL);
+        cl_int err = clEnqueueFillBuffer(data->handle->cmdq, data->bnBias, &pattern, sizeof(cl_float), 0, input_dims[2], 0, NULL, NULL);
```
Comment (rgiduthuri) [inline vx_nn/src/scale_layer.cpp]: When parameter[4] is not specified, there will be a memory leak as there is no matching release corresponding to clCreateBuffer.

Comment (rgiduthuri) [reply vx_nn/src/scale_layer.cpp]: Meant parameter[2]... copy-paste error

File: vx_nn/src/scale_layer.cpp [hunk 3]
```diff
@@ -150,6 +150,9 @@ static vx_status VX_CALLBACK uninitializeScaleLayer(vx_node node, const vx_refer
     ERROR_CHECK_MIOPEN_STATUS(miopenDestroyTensorDescriptor(data->input_desc));
     ERROR_CHECK_MIOPEN_STATUS(miopenDestroyTensorDescriptor(data->output_desc));
     ERROR_CHECK_MIOPEN_STATUS(miopenDestroyTensorDescriptor(data->bnScaleBiasMeanVarDesc));
+    if(!parameters[4]){
+     clReleaseMemObject(data->bnBias);
```
Comment (kiritigowda) [inline vx_nn/src/scale_layer.cpp]: clmem release.

File: vx_nn/src/scale_layer.cpp (hunk 2 above)
Comment (rgiduthuri) [reply vx_nn/src/scale_layer.cpp]: Also need error check for clCreateBuffer

File: vx_nn/src/scale_layer.cpp (hunk 3 above)
Comment (rgiduthuri) [reply vx_nn/src/scale_layer.cpp]: Need a check for nullptr before release

**Pull request merged**